### PR TITLE
Endpoint teams

### PR DESCRIPTION
**Pull request by mixman**

### CODE REVIEW

File: src/PlanMill/EndPoints/Teams.hs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    , tCostCenter     :: !(Maybe Int)
+    , tCreated        :: !(Maybe UTCTime)
+    , tCreatedBy      :: !(Maybe Int)
+    , tDescription    :: !(Maybe Text)
+    , tLatestMember   :: !(Maybe Int)
+    , tManager        :: !(Maybe Int)
+    , tMembers        :: !(Maybe Int)
+    , tModified       :: !(Maybe UTCTime)
+    , tModifiedBy     :: !(Maybe Int)
+    , tParentTeam     :: !(Maybe Int)
```
Comment (phadej) [inline src/PlanMill/EndPoints/Teams.hs]: `Maybe TeamId`

File: src/PlanMill/EndPoints/Teams.hs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+data Team = Team
+    { tId             :: !TeamId
+    , tName           :: !(Maybe Text)
+    , tChildTeams     :: !(Maybe Int)
+    , tCostCenter     :: !(Maybe Int)
+    , tCreated        :: !(Maybe UTCTime)
+    , tCreatedBy      :: !(Maybe Int)
+    , tDescription    :: !(Maybe Text)
+    , tLatestMember   :: !(Maybe Int)
+    , tManager        :: !(Maybe Int)
```
Comment (phadej) [inline src/PlanMill/EndPoints/Teams.hs]: `Maybe UserId`

File: src/PlanMill/EndPoints/Teams.hs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+team i = planMillGet $ t "teams" // i
+  where t = id :: Text -> Text
+
+data Team = Team
+    { tId             :: !TeamId
+    , tName           :: !(Maybe Text)
+    , tChildTeams     :: !(Maybe Int)
+    , tCostCenter     :: !(Maybe Int)
+    , tCreated        :: !(Maybe UTCTime)
+    , tCreatedBy      :: !(Maybe Int)
```
Comment (phadej) [inline src/PlanMill/EndPoints/Teams.hs]: `Maybe UserId`

File: src/PlanMill/EndPoints/Teams.hs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+-- | A single team in PlanMill
+--
+-- See <https://online.planmill.com/pmtrial/schemas/v1_5/index.html#teams__id__get>
+team :: TeamId -> PlanMill Team
+team i = planMillGet $ t "teams" // i
+  where t = id :: Text -> Text
+
+data Team = Team
+    { tId             :: !TeamId
+    , tName           :: !(Maybe Text)
```
Comment (phadej) [inline src/PlanMill/EndPoints/Teams.hs]: Here as well, every field seems to be `Maybe`. I'd expect at least `name` to be mandatory for teams.

File: src/PlanMill/EndPoints/Teams.hs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    , tManager        :: !(Maybe Int)
+    , tMembers        :: !(Maybe Int)
+    , tModified       :: !(Maybe UTCTime)
+    , tModifiedBy     :: !(Maybe Int)
+    , tParentTeam     :: !(Maybe Int)
+    , tParentTeamName :: !(Maybe Text)
+    -- team/ID
+    , tParent         :: !(Maybe Int)
+    , tPassive        :: !(Maybe Int)
+    , tOwner          :: !(Maybe Int)
```
Comment (phadej) [inline src/PlanMill/EndPoints/Teams.hs]: this is `UserId` as well, isn't it?

File: src/PlanMill/EndPoints/Teams.hs (hunk 1 above)
Comment (mixman) [inline src/PlanMill/EndPoints/Teams.hs]: tId always exists?

Comment (mixman) [inline src/PlanMill/EndPoints/Teams.hs]: ah, nvm. i can't eye where github comments refer to yet!

**Pull request merged**